File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)

{
    string s = get_string("Text: ");

    int letters = count_letters(s);
    int words = count_words(s);
    int sentences = count_sentences(s);
    float index;

//Convert the return to float to help the division
    float st = 100 * (float)sentences / (float)words ; // Gets the average sentences per 100 words
    float l = 100 * (float)letters / (float)words;  //Gets the average letters per 100 words

    index = 0.0588 * l - 0.296 * st - 15.8;

    //Use the math.h method to round the index
    int  indexr = round(index);

    if (indexr >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (indexr < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", indexr);
    }




}
int count_letters(string text)
{
    int countL = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        //Check each character if is  alphabetical
        if (isalpha(text[i]))
        {
            countL++;
        }

        
    }
    return countL;
}
int count_words(string text)
{
    int countW = 1; // Assuming will always have at least one word

    for (int i = 0; i < strlen(text) ; i++)

    {
        // Check if has spaces between the text, and gets word counter by 1
        if (text[i] == ' ')
        {
            countW++;
        }
    }

    return countW;
}
int count_sentences(string text)
{
    int countST = 0;
    for (int i = 0; i < strlen(text) ; i++)

    {
        //If the text has one of these, the sentence counter goes by 1
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            countST++;
        }
    }
    return countST;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    s = get_string("Insert String: ")

    letters = count_letters(s)
    words = count_words(s)
    sentences = count_sentences(s)

    index = 0

    st = 100 * float(sentences) / float(words)
    l = 100 * float(letters) / float(words)

    index = 0.0588 * l - 0.296 * st - 15.8

    # round index

    indexr = round(index)

    if (indexr >= 16):
        print("Grade 16+")

    elif (indexr < 1):
        print("Before Grade 1")

    else:
        print(f"Grade {indexr}")  # Using string interpolation


def count_letters(text):
    countL = 0
    for char in text:
        if (char.isalpha()):  # Check if each character is alphabetical
            countL += 1
    return countL


def count_words(text):
    countW = 1  # Assuming will always have one word

    for char in text:
        if (char.count(" ")):  # using count functin to look out for spaces
            countW += 1
    return countW


def count_sentences(text):
    countST = 0

    for char in text:
        if (char == "!" or char == "?" or char == "."):  # check if each character of the string is an sentence
            countST += 1
    return countST


main()